src/main/scala/mesosphere/marathon/core/volume/providers/DVDI.scala

```diff
@@ -0,0 +1,220 @@
+package mesosphere.marathon.core.volume.providers
+
+import com.wix.accord._
+import com.wix.accord.dsl._
+import com.wix.accord.Validator
+import mesosphere.marathon.core.volume._
+import mesosphere.marathon.state._
+import org.apache.mesos.Protos.{ ContainerInfo, Volume => MesosVolume, Environment }
+import scala.collection.JavaConverters._
+
+/**
+  * DVDIProvider (Docker Volume Driver Interface provider) handles persistent volumes allocated
+  * by a specific docker volume driver plugin. This works for both docker and mesos containerizers,
+  * albeit with some limitations:
+  *   - only a single volume driver per container is allowed when using the docker containerizer
+  *   - docker containerizer requires that referenced volumes be created prior to application launch
+  *   - mesos containerizer only supports volumes mounted in RW mode
+  */
+protected case object DVDIProvider extends InjectionHelper[PersistentVolume]
+    with PersistentVolumeProvider with OptionSupport {
+
+  import org.apache.mesos.Protos.Volume.Mode
+
+  val NAME = "dvdi"
+  override val name = Some(NAME)
+
+  sealed trait OptNS { val namespace: String = NAME }
```

Comment:
Any reason why you use traits as base class of case classes? Overriding name below feels strange.